components/atom/backToTop/src/index.js

```diff
@@ -16,95 +18,80 @@ const STYLES = {
   LIGHT: 'light'
 }
 
-class AtomBackToTop extends PureComponent {
-  state = {
-    show: null,
-    hover: false
-  }
-
-  intervalId = 0
-  container = null
-
-  scrollStep = () => {
-    const {scrollTop} = this.container
-    const {scrollSteps} = this.props
-    const {intervalId} = this
-
-    if (scrollTop === 0) clearInterval(intervalId)
-    this.container.scrollTop = scrollTop - scrollSteps
-  }
-
-  scrollToTop = () => {
-    const {scrollStep} = this
-    const {scrollIntervalTime} = this.props
-    this.intervalId = setInterval(scrollStep, scrollIntervalTime)
-  }
-
-  handleScroll = halfHeight => {
-    const {scrollTop} = this.container
-    const {show} = this.state
+const AtomBackToTop = ({
+  iconTop: IconTop,
+  textTop,
+  scrollSteps,
+  scrollIntervalTime,
+  refContainer,
+  style
+}) => {
+  const [show, setShow] = useState(false)
+  const [hover, setHover] = useState(false)
+
+  let intervalId = 0
```

Comment:
I think this should be a `ref` otherwhise, the value will be last in each render.

Reply:
Well, the `ref` is actually being passed via `refContainer` prop.

Reply:
I mean the `intervalId` value that it might be a ref.

```
const intervalId = useRef()
```

Otherwise, the value could be recreated for every render and you will lose the value.

Reply:
oks, fixed!